parquet-common/src/main/java/org/apache/parquet/bytes/SingleBufferInputStream.java

```diff
@@ -174,4 +248,63 @@ public boolean markSupported() {
   public int available() {
     return buffer.remaining();
   }
+
+  @Override
+  public byte readByte() throws IOException {
```

Comment:
The changes from here on out look like what you're really trying to do because we want to read directly from the stream. Can you remove all the other changes that aren't needed?

Reply:
I'm not sure what you're referring to. All of the methods beyond this point are absolutely necessary. We need to be able to read ints and longs and such from the bytebuffer, and this is the only way to get them.